lib/segment/src/index/hnsw_index/graph_layers_builder.rs

```diff
@@ -403,14 +403,42 @@ impl GraphLayersBuilder {
             // New point is a new empty entry (for this filter, at least)
             // We can't do much here, so just quit
         }
-        self.ready_list.write().set(point_id as usize, true);
+        let was_ready = self.ready_list.write().replace(point_id as usize, true);
+        assert!(!was_ready);
```

Comment:
debug assert or do we want to crash here?

Reply:
Replaced with `debug_assert`.